src/consensus/tx_verify.h

```diff
@@ -24,7 +24,7 @@ namespace Consensus {
  * @param[out] txfee Set to the transaction fee if successful.
  * Preconditions: tx.IsCoinBase() is false.
  */
-bool CheckTxInputs(const CTransaction& tx, TxValidationState& state, const CCoinsViewCache& inputs, int nSpendHeight, CAmount& txfee);
+[[nodiscard]] bool CheckTxInputs(const CTransaction& tx, TxValidationState& state, const CCoinsViewCache& inputs, int nSpendHeight, CAmount& txfee);
```

Comment:
might be nice to change this to a struct similar to `struct MempoolAcceptResult`, so that `std::optional` can be used

Reply:
Agreed! That would make a nice follow-up PR. Keeping this PR as-is to limit scope.